code/game/gamemodes/objective.dm

```diff
@@ -616,6 +616,19 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return FALSE
 	return TRUE
 
+/datum/objective/destroy/post_target_cryo(list/owners)
+	var/datum/objective/new_objective
+	for(var/datum/mind/M in owners)
+		new_objective = new /datum/objective/assassinate
+		new_objective.owner = M
+		new_objective.find_target()
+		if(!new_objective.target)
+			qdel(new_objective)
+			continue
+		else
+			M.objectives += new_objective
+		M.remove_objective(src)
```

Comment:
with #22374, this will be made a lot easier to do and be less scuffed. HC/CA may want to wait until then so that this can be a less scuffed fix, because M.remove_objective(src) will start qdeling it at the end of the first loop

Reply:
TBH 22374 might take a while to get in, and we have no TM support right now ☹️. In the mean time we could just perform a second loop after everything in the first is done, and simply call `M.remove_objective(src)`. Not ideal, but yeah.

Reply:
That sounds like a better solution for now. If this gets merged before 22374 (it likely will). I can update it in my PR